File: TestLuaj.java
import org.luaj.LuaTable;
import org.luaj.LuaValue;
import org.luaj.lib.VarArgFunction;
import org.luaj.Varargs;

public class TestLuaj {
    public static void main(String[] args) {
        LuaTable mt = new LuaTable();
        mt.set("__add", new VarArgFunction() {
            @Override
            public Varargs invoke(Varargs args) {
                System.out.println("__add invoked!");
                return LuaValue.valueOf("added!");
            }
        });
        
        LuaTable t1 = new LuaTable();
        t1.setmetatable(mt);
        LuaTable t2 = new LuaTable();
        t2.setmetatable(mt);
        
        LuaValue result = t1.add(t2);
        System.out.println("Result of t1 + t2: " + result);
    }
}

File: test_luaj.kt
import org.luaj.LuaTable
import org.luaj.LuaFunction
import org.luaj.LuaValue
import org.luaj.lib.VarArgFunction
import org.luaj.Varargs

fun main() {
    val mt = LuaTable()
    mt.set("__add", object : VarArgFunction() {
        override fun invoke(args: Varargs): Varargs {
            println("__add invoked!")
            return LuaValue.valueOf("added!")
        }
    })
    
    val t1 = LuaTable()
    t1.setmetatable(mt)
    val t2 = LuaTable()
    t2.setmetatable(mt)
    
    val result = t1.add(t2)
    println("Result of t1 + t2: " + result)
}
